.github/workflows/auto-deployments.yml

```diff
@@ -0,0 +1,89 @@
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Check changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v32.1.2
+        with:
+          dir_names: true
+          json: true
+
+      - id: set-matrix
+        if: contains(steps.changed-files.outputs.all_changed_files, '-spa')
+        run: echo "matrix={\"container\":${{ steps.changed-files.outputs.all_changed_files }}}" >> "$GITHUB_OUTPUT"
+
+  matrix-job:
+    name: Job for ${{ matrix.container }}
+    runs-on: self-hosted
+    needs: [get-changes-for-spas]
+    strategy:
+      matrix: ${{ fromJSON(needs.get-changes-for-spas.outputs.matrix) }}
+      max-parallel: 4
+      fail-fast: false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        if: contains(matrix.container, '-spa')
+      - name: Use Node.js 16.x
+        uses: actions/setup-node@v3
+        if: contains(matrix.container, '-spa')
+        with:
+          node-version: 16
+          cache: 'npm'
+          cache-dependency-path: 'packages/**/package-lock.json'
+      - name: Get SPA NAME as strings
+        id: get-spa-name
+        if: contains(matrix.container, '-spa')
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const spaName = "${{ matrix.container }}".split('/')[1];
+            core.setOutput('spa-name', spaName)
+      - name: Build and deploy SPA on DEV
+        if: contains(matrix.container, '-spa')
+        env:
+          SPASHIP_DEPLOY_URL: "${{ secrets.SPASHIP_DEPLOY_URL }}dev"
+        # Steps in the following script
+        # #1 cd into dir
+        # #2 install the required node_modules
+        # #3 get env file from gitlab (note: without the full url curl does not resolve the endpoint)
+        # #4 build with the .env file
+        # #5 deploy using spaship cli
+        run: |
+          cd ${{ matrix.container }}
+          npm install
+          curl -k --header "PRIVATE-TOKEN:${{ secrets.GITLAB_TOKEN }}" "https://gitlab.cee.redhat.com/api/v4/projects/9092/repository/files/spa-configs%2Fdev%2F${{ steps.get-spa-name.outputs.spa-name }}%2F.env/raw?ref=master" -o .env
+          npm run build
+          export VERSION=\$(node -p "require('./package.json').version")
+          npx @spaship/cli@latest deploy --apikey=${{ secrets.SPASHIP_API_KEY }} \
+          --env=$SPASHIP_DEPLOY_URL \
+          --ref=\$VERSION
+      - name: Build and deploy SPA on QA
+        if: contains(matrix.container, '-spa')
+        env:
+          SPASHIP_DEPLOY_URL: "${{ secrets.SPASHIP_DEPLOY_URL }}qa"
+        run: |
+          cd ${{ matrix.container }}
+          npm install
+          curl -k --header "PRIVATE-TOKEN:${{ secrets.GITLAB_TOKEN }}" ""https://gitlab.cee.redhat.com/api/v4/projects/9092/repository/files/spa-configs%2Fqa%2F${{ steps.get-spa-name.outputs.spa-name }}%2F.env/raw?ref=master" -o .env
```

Comment:
Can you use this in the environments because Its internal URL can you use in the environment variables

Reply:
I did try using github secrets which didn't work, let me try again and check if it's possible

Reply:
for example if you see this commit https://github.com/1-Platform/one-platform/pull/1642/commits/737a790035eee8a067ac9c1d5f05f29e1057c892 it was saved in github secrets and didn't work

Reply:
Resolving as @riginoommen is on a PTO